packages/workspaces-extension/src/index.ts

```diff
@@ -46,13 +48,41 @@ const workspacesMenu: JupyterFrontEndPlugin<void> = {
   }
 };
 
+/**
+ * The extension providing workspace indicator at topbar
+ */
+const workspacesIndicator: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlab/workspaces-extension:indicator',
+  description: 'Adds a worspace indicator and selector element at top',
+  requires: [IWorkspacesModel, IWorkspaceCommands, IWindowResolver],
+  autoStart: true,
+  activate: (
+    app: JupyterFrontEnd,
+    model: IWorkspacesModel,
+    commands: IWorkspaceCommands,
+    resolver: IWindowResolver
+  ) => {
+    const openWorkspace = (workspace: string) => {
+      app.commands.execute(commands.open, { workspace: workspace });
+    };
+    const widget = new WorkspaceSelectorWidget({
+      currentWorkspace: resolver.name,
+      identifiers: model.identifiers,
+      openWorkspace: openWorkspace,
+      model: model
+    });
+    app.shell.add(widget, 'top', { rank: 1000 });
```

Comment:
Wondering if it should be placed in the top bar by default, or made opt-in? Via a setting or a toggle in the running panel.

Not sure workspaces is a feature used by many users, and a new dropdown in the top area may take significant space and add distraction?

Reply:
So the thinking is that many users find that their tabs "are gone" if they open JupyterLab in a new window. This is because they were routed to another workspace. Exposing non-default workspace would allow them to recover their workspace easily.  In a sense this would be an important distraction. However, I agree that if we are showing it as opt-out it should occupy much less space.

Maybe we should add the it as opt-in for 4.4 and decide on whether to make it opt-out for next release later?

Reply:
> Maybe we should add the it as opt-in for 4.4 and decide on whether to make it opt-out for next release later?

Sounds good.

It's often difficult to decide what should be enabled by default when we lack metrics and usage data. My personal feeling was that workspaces are used by a handful of users only, but this may be a wrong statement.